doc/oauth-guide.md

```diff
@@ -0,0 +1,65 @@
+# Using 3scale API Gateway with OAuth
+
+## Pre-requisites
+
+APICast offers support for the OAuth Authorization Code flow out of the box as long as the following pre-requisites are met:
+
+- An Authorization Server as defined in [RFC6749#1.1](https://tools.ietf.org/html/rfc6749#section-1.1) with the one exception that the access tokens will be issued by APIcast instead. In this case the Authorization Server will only authenticate the resource owner and obtain their authorization.
+- A Redis instance.
+
+## APIcast Configuration in 3scale Admin Portal
+
+In order to configure this, we will first need to choose the OAuth Authentication method from the Integration Settings (API > Integration > "edit integration settings" ) screen on our 3scale Admin portal. We will also need to ensure we have selected the Self Managed Gateway option, as it is not currently possible to run APIcast with OAuth Authorization Code support on the APIcast cloud gateway. 
+
+Once that is done, we will see an additional field in the Integration Screen ( API > Integration ) under "Authentication Settings": "OAuth Authorization Endpoint." Here we define where Resource Owners will be redirected to in order to authenticate and confirm they authorize a given client access to their resources ( as per [RFC6749#4.1](https://tools.ietf.org/html/rfc6749#section-4.1.1).) 
+
+All other fields on the Integration Screen should be configured as per the "APIcast Overview" document. Since we will be running all components locally for this example, my "Public Base URL" where APIcast is running will be `http://localhost:8080`
+
+(Screenshot)
```

Comment:
I guess we will have to add screenshots? 

I'd incline to have minimal working configuration as JSON that they can try with and edit one field to get it working rather than document the Admin UI. Imo this belongs to the product, not to the upstream gateway.